### PR TITLE
sauces up angle2dir with some #define red bull

### PR DESCRIPTION
**Pull request by vuonojenmustaturska**
Clipboard05

9ffe9f45a9326f03

### CODE REVIEW

**Comment by Qustinnus** [comment]
Welcome back king

**Comment by Couls** [comment]
I like your shirt

**Comment by vuonojenmustaturska** [comment]
image

**Comment by spookydonut** [comment]
weird math voodoo

File: code/__HELPERS/type2type.dm [hunk 1]
```diff
@@ -120,27 +120,8 @@
...
-		if(247.5 to 292.5)
-			return WEST
-		if(292.5 to 337.5)
-			return NORTHWEST
-		if(337.5 to 360)
-			return NORTH
+GLOBAL_LIST_INIT(modulo_angle_to_dir, list(NORTH,NORTHEAST,EAST,SOUTHEAST,SOUTH,SOUTHWEST,WEST,NORTHWEST))
+#define angle2dir(X) (GLOB.modulo_angle_to_dir[round((((X%360)+382.5)%360)/45)+1])
 
 /proc/angle2dir_cardinal(degree)
```
Comment (Dennok) [inline code/__HELPERS/type2type.dm]: Why don't convert this too?

File: code/__HELPERS/type2type.dm [hunk 2]
```diff
@@ -120,27 +120,8 @@
...
-			return SOUTH
-		if(202.5 to 247.5)
-			return SOUTHWEST
-		if(247.5 to 292.5)
-			return WEST
-		if(292.5 to 337.5)
-			return NORTHWEST
-		if(337.5 to 360)
-			return NORTH
+GLOBAL_LIST_INIT(modulo_angle_to_dir, list(NORTH,NORTHEAST,EAST,SOUTHEAST,SOUTH,SOUTHWEST,WEST,NORTHWEST))
```
Comment (Dennok) [inline code/__HELPERS/type2type.dm]: why not change `alldirs` to this and use `alldirs` here?

**Pull request merged**